src/TwigComponent/src/EventListener/PreRenderEvent.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent\EventListener;
+
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+final class PreRenderEvent extends Event
+{
+    public function __construct(
+        public object $component,
+        public string $template,
+        public array $context,
+        public array $config
+    ) {
```

Comment:
I made `template` and `context` (which is the vars sent to the template) public because they have the option to me changed. So these would need setters. I have been trying not to create _anemic_ getters/setters for these type of properties but can if you'd like. Getters/setters _would_ give the ability to better document "what they do".

For config, private with just a getter makes sense as this value is not meant to be changed.

Reply:
To further on my explanation, manipulating say `$context`, would be more verbose to manipulate:

```php
$context = $event->getContext();
$context['foo'] = 'bar';
$event->setContext($context);

// opposed to
$event->context['foo'] = 'bar';
```

Additionally, in PHP 8.1, I'd make `$config` readonly.

This all being said, if you think we should follow established patterns, I'm totally fine with that.

Reply:
Bah, no strong opinion and, obviously, things are changing with newer php versions. I think I'd like to stay "boring" for the time being. For context, (or variables, whatever we call it), we could also have a `addVariable()` method. This is all kind of academic, since this event won't be seen by many.